LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -6,58 +6,84 @@ namespace LearningMissionSimulation
+        List<Instructor> _instructorList = new List<Instructor>();
+        Dictionary<Guid, Module> _moduleDictionary = new Dictionary<Guid, Module>();
+        Dictionary<Guid, Classroom> _classroomDictionary = new Dictionary<Guid, Classroom>();
+        Dictionary<Guid, List<Instructor>> _moduleInstructorListDictionary = new Dictionary<Guid, List<Instructor>>();
+        #endregion Fields
 
-        Dictionary<Guid, Account> accountDictionary = new Dictionary<Guid, Account>();
-        Dictionary<Guid, Instructor> instructorDictionary = new Dictionary<Guid, Instructor>();
-        Dictionary<Guid, Student> studentDictionary = new Dictionary<Guid, Student>();
-        List<Account> pendingAccountList = new List<Account>();
-        List<Subject> subjectList = new List<Subject>();
-        List<Guid> subjectIdList = new List<Guid>();
-        List<Guid> moduleIdList = new List<Guid>();
-        List<Student> studentList = new List<Student>();
-        List<Instructor> instructorList = new List<Instructor>();
-        Dictionary<Guid, Module> moduleDictionary = new Dictionary<Guid, Module>();
+        #region Properties
+        public Dictionary<Guid, Account> AccountDictionary { get => _accountDictionary; set => _accountDictionary = value; }
+        public Dictionary<Guid, Instructor> InstructorDictionary { get => _instructorDictionary; set => _instructorDictionary = value; }
+        public List<Student> ActiveStudentList { get => _activeStudentList; set => _activeStudentList = value; }
+        public Dictionary<Guid, Student> StudentDictionary { get => _studentDictionary; set => _studentDictionary = value; }
+        public List<Instructor> ActiveInstructorList { get => _activeInstructorList; set => _activeInstructorList = value; }
+        public List<Account> PendingAccountList { get => _pendingAccountList; set => _pendingAccountList = value; }
+        public List<Subject> SubjectList { get => _subjectList; set => _subjectList = value; }
+        public List<Guid> SubjectIdList { get => _subjectIdList; set => _subjectIdList = value; }
+        public List<Guid> ModuleIdList { get => _moduleIdList; set => _moduleIdList = value; }
+        public List<Student> StudentList { get => _studentList; set => _studentList = value; }
+        public List<Instructor> InstructorList { get => _instructorList; set => _instructorList = value; }
+        public Dictionary<Guid, Module> ModuleDictionary { get => _moduleDictionary; set => _moduleDictionary = value; }
+        public Dictionary<Guid, Classroom> ClassroomDictionary { get => _classroomDictionary; set => _classroomDictionary = value; }
+        public Dictionary<Guid, List<Instructor>> ModuleInstructorListDictionary { get => _moduleInstructorListDictionary; set => _moduleInstructorListDictionary = value; }
+        #endregion Properties
 
         public void CreateAccounts(int accountCount)
         {
+            ReportHeader("Create account");
```

Comment:
Add a local variable for the action:
string action = "Create accounts"
Do the same in other methods, the action should be the name of the method:
For ActivateAccounts: action="Activate accounts"
For AssignModulesToStudents: action="Assign modules to students"
and so on.
Then you can use it with all report methods: ReportHeader(action); ReportFooter(action);
ReportSummary(actionName, itemName, itemCount);